wandb/sdk_py27/wandb_artifacts.py

```diff
@@ -472,7 +479,11 @@ def add(self, obj, name):
         entry = self._manifest.get_entry_by_path(name)
         if entry is not None:
             return entry
-        with self.new_file(name) as f:
+        file_path = os.path.join(ARTIFACT_TMP.name, str(id(self)), name)
```

Comment:
Does this mean we're going to write a bunch of files to wherever the temp directory happens to be and leave them there?

I believe maybe we're also doing this with the fixes Anni put in for the cache race.

Just be aware of the potential to blow up wherever the temp file system is with too much data. User's will probably want control of where this happens if there's any risk of that.

Reply:
added explicit cleanup